aten/src/ATen/native_parse.py

```diff
@@ -186,6 +186,83 @@ def parse_arguments(args, func_variants, declaration, func_return):
     if is_out_fn:
         declaration['name'] += "_out"
 
+    # Reverse splat of TensorOptions
+    # As we move towards the JIT function schema for native_functions.yaml we need to support
+    # the expanded version of TensorOptions. For this we discover whether there are three
+    # types and names of keyword arguments: "ScalarType dtype", "Layout layout" and "Device device"
+    # Each, if set,  must have default arguments set to long or float, strided and "cpu" respectively.
+    # They must appear in this order and in this order only to be compliant with the JIT schema
+
+    tensor_options_arguments = [
+        # XXX - until we actually have first-class interpreter types for these
+        # concepts, the default values to be encoded in Tensors
+        # If you change this, you also need to update [TensorOptions in script]
+        # in the tracer code.
+        # dtype is specified as an int64_t of at::ScalarType
+        {'name': 'dtype', 'type': ['ScalarType', 'ScalarType?'], 'is_nullable': False, 'annotation': None},
+        # layout is specified as an int64_t of at::Layout
+        {'name': 'layout', 'type': ['Layout', 'Layout?'], 'is_nullable': False, 'annotation': None},
+        # device is specified as an IntArrayRef of { at::Device::Type, device_id }
+        {'name': 'device', 'type': ['Device', 'Device?'], 'is_nullable': False, 'annotation': None},
+    ]
+    tensor_options_defaults = [['c10::nullopt'], ['c10::nullopt'], ['c10::nullopt']]
+
+    def compare_tensor_option(argument, tensor_option_argument):
+        matches = True
+        matches = matches and argument['name'] == tensor_option_argument['name']
+        # print("--")
+        # print("declaration['name']")
+        # print(declaration['name'])
+        # print("tensor_option_argument['type']")
+        # print(tensor_option_argument['type'])
+        # print("argument['type']")
+        # print(argument['type'])
+        matches = matches and argument['type'] in tensor_option_argument['type']
+        return matches
+
+    def is_tensor_option(argument, tensor_options_arguments):
+        is_in_it = False
+        for to_argument in tensor_options_arguments:
+            is_in_it = is_in_it or compare_tensor_option(argument, to_argument)
+        return is_in_it
+
+    new_arguments = []
+    idx = 0
+    while idx < len(arguments):
+        argument = arguments[idx]
+        if is_tensor_option(argument, tensor_options_arguments) and \
+                len(arguments) - idx >= 3:
+            tensor_options_representation = []
```

Comment:
do we really need all the generic code below here?  Since this should go away anyway, can we just literally match up the signature, e.g. just have a dict from the string to the TensorOptions definition?

i.e. something like

dict(
   "ScalarType? dtype=None, Layout? layout=None, Device? device=None": {'type': 'TensorOptions', 'name:', ...)

you'd only have like ~5 entries right?

And you could just say "not supported combination of TensorOptions, the only currently supported combinations are", dict.keys().